README.md

```diff
@@ -1,234 +1,679 @@
+- **`$TASK_NAME`**: A parameter to separate different tasks (this value may include `/` characters, creating nested directories).
+
+The file glob `glob($TASK_ROOT/$TASK_NAME/**/*.parquet)` is used to capture all task label files. Note that:
+
+- The sharding of task label files may differ from that of the raw data files.
+- In some cases, a shard may not contain any task labels if no subject qualifies for the task; such files might be empty or missing.
+
+## Example: MIMIC-IV demo dataset
+
+Let's look at the publicly available [MIMIC-IV demo](https://physionet.org/content/mimic-iv-demo/2.2/) dataset
+for a concrete example of what a MEDS-compliant dataset may look like. We use [MIMIC_IV_MEDS](https://github.com/Medical-Event-Data-Standard/MIMIC_IV_MEDS) ETL pipeline to automatically download the MIMIC-IV demo dataset and convert it into MEDS format. After running the pipeline, we get a directory structure that looks like this:
+
+```console
+MEDS_cohort
+├── ...
+├── data
+│   ├── held_out
+│   │   └── 0.parquet
+│   ├── train
+│   │   └── 0.parquet
+│   └── tuning
+│       └── 0.parquet
+├── ...
+├── metadata
+│   ├── codes.parquet
+│   ├── dataset.json
+│   └── subject_splits.parquet
+└── ...
+```
+
+We can immediately see that the data is organized as described in the [Organization on Disk](#organization-on-disk) section, with the `data` directory containing the raw underlying event data and the `metadata` directory containing the metadata. The `data` directory contains three subdirectories, `train`, `tuning`, and `held_out`, corresponding to our default split names and each containing a data file in parquet format (since the MIMIC-IV demo dataset is small, we only have one data file per split but in general we would expect multiple shards).
+
+Let's take a look at the contents of one of the data files in the `train` split, which looks something like this (note that we are using the polars package for better readability):
+
+```python
+from pyarrow import parquet as pq
+import polars as pl
+
+data_tbl = pq.read_table("data/train/0.parquet")
+pl.from_arrow(data_tbl)
+```
+
+```console
+shape: (803_992, 25)
+┌────────────┬─────────────────────┬──────────────────────┬───────────────┬────────────┬───┐
+│ subject_id ┆ time                ┆ code                 ┆ numeric_value ┆ text_value ┆ … │
+│ ---        ┆ ---                 ┆ ---                  ┆ ---           ┆ ---        ┆   │
+│ i64        ┆ datetime[μs]        ┆ str                  ┆ f32           ┆ str        ┆   │
+╞════════════╪═════════════════════╪══════════════════════╪═══════════════╪════════════╪═══╡
+│ 12345678   ┆ null                ┆ GENDER//M            ┆ null          ┆ null       ┆ … │
+│ 12345678   ┆ 2138-01-01 00:00:00 ┆ MEDS_BIRTH           ┆ null          ┆ null       ┆ … │
+│ 12345678   ┆ 2178-02-12 08:11:00 ┆ LAB//51079//UNK      ┆ null          ┆ POS        ┆ … │
+│ 12345678   ┆ 2178-02-12 11:38:00 ┆ LAB//51237//UNK      ┆ 1.4           ┆ null       ┆ … │
```

Comment:
Just flagging here that I originally made these rows up because despite `mimic-iv-demo` being public, I wasn't sure what the policy was on actually displaying individual data here. We can keep it made up, place a disclaimer, or replace with actualy `mimic-iv-demo` data.